src/main/scala/uk/gov/hmrc/SbtArtifactory.scala

```diff
@@ -136,15 +137,17 @@ object SbtArtifactory extends sbt.AutoPlugin{
         }
       }
     }).value,
-    publishToBintray := {
+    publishToAll := publishToBintray.dependsOn(publish).value,
+    publishAndDistribute := {
       sLog.value.info(s"SbtArtifactoryPlugin - 'publishToBintray' is deprecated. Please use publishToAll instead")
       publishToAll.value
-    },
-    publishAndDistribute := publishToBintray.dependsOn(publish).value
+    }
   )
 
   override def trigger = allRequirements
 
+  override def requires: Plugins = BintrayPlugin
```

Comment:
Maybe add a comment, that without this, the BintrayPlugin#bintrayRepository default definition may be applied after our definition.